### PR TITLE
Migrate original provenance verifier

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

cc @laurentsimon

### CODE REVIEW

**Comment by laurentsimon** [comment]
Can we remove all the testdata that are for the workflow verification?

**Review by laurentsimon — APPROVED** [review]

File: pkg/provenance.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+		return nil, errors.New("inclusion proof not provided")
+	}
+
+	hashes := [][]byte{}
+	for _, h := range e.Verification.InclusionProof.Hashes {
+		hb, _ := hex.DecodeString(h)
+		hashes = append(hashes, hb)
+	}
+
+	rootHash, _ := hex.DecodeString(*e.Verification.InclusionProof.RootHash)
```
Comment (laurentsimon) [inline pkg/provenance.go:123]: why not verify the hash?

**Review by laurentsimon — APPROVED** [review]

File: pkg/provenance.go (hunk 1 above)
Comment (asraa) [reply pkg/provenance.go:123]: Interesting. I copied this from cosign's implementation, but I think you are right. I don't think this, or cosign, verifies the signature on the tree head/root hash... going to talk offline :o

Comment (laurentsimon) [reply pkg/provenance.go:123]: also, verify the returned error. I'm not sure was it's intentionally discarded in cosign code

Comment (asraa) [reply pkg/provenance.go]: added both verifications

**Comment by asraa** [comment]
DO NOT MERGE.

There's problems with Rekor's loginfo when I added the root hash verification. LogInfo might return the root hash for *any* one of the instances, I think, without a UUID identifier.

**Comment by asraa** [comment]
Done!

**Review by laurentsimon — APPROVED** [review]
Thanks!

**Pull request merged**